server/devdocs/bin/generate-proptypes-index.js

```diff
@@ -4,31 +4,99 @@
  * This file generates an index of proptypes by component displayname, slug and folder name
  */
 
+const startTime = process.hrtime();
+
 /**
  * External Dependencies
  */
-
+require( 'babel-register' );
 const fs = require( 'fs' );
 const path = require( 'path' );
 const reactDocgen = require( 'react-docgen' );
+const { getPropertyName, getMemberValuePath, resolveToValue } = require( 'react-docgen/dist/utils' );
+const util = require( 'client/devdocs/docs-example/util' );
 
 const root = path.dirname( path.join( __dirname, '..', '..' ) );
-const pathSwap = new RegExp(path.sep, 'g');
+const pathSwap = new RegExp( path.sep, 'g' );
+const handlers = [ ...reactDocgen.defaultHandlers, commentHandler ];
 
 /**
- * Converts a camel cased string into a slug
- * @param {String} name The camel cased string to slugify
- * @return {String}
+ * Replaces **'s in comment blocks and trims comments
+ * @param {string} str The doc string to clean
+ * @return {string} The clean comment block text
  */
-const camelCaseToSlug = ( name ) => {
-	if ( ! name ) {
-		return name;
+function parseDocblock( str ) {
+	const lines = str.split( '\n' );
+	for ( let i = 0, l = lines.length; i < l; i++ ) {
+		lines[ i ] = lines[ i ].replace( /^\s*\*\s?/, '' );
 	}
+	return lines.join( '\n' ).trim();
+}
 
-	return name
-		.replace( /\.?([A-Z])/g, ( x, y ) => '-' + y.toLowerCase() )
-		.replace( /^-/, '' );
-};
+/**
+ * Given a path, this function returns the closest preceding comment, if it exists
+ * @param {NodePath} nodePath The node path from react-docgen
+ * @return {null|string} The comment or nothing, if no comment
+ */
+function getComments( nodePath ) {
+	let comments = [];
+
+	if ( nodePath.node.leadingComments ) {
+		// if there are comments before this property node, use the ones leading, not following a previous node
+		comments = nodePath.node.leadingComments.filter(
+			comment => comment.leading === true
+		);
+	} else if ( nodePath.node.comments ) {
+		// if there are comments after this property node, use the ones following this node
+		comments = nodePath.node.comments.filter(
+			comment => comment.leading === false
+		);
+	}
```

Comment:
I don't see how this could work. `leadingComments` is an array of leading comments. Such that:

``` js
do.something(); // this is a leading comment to the next line, but will have comment.leading set to false
// this is a leading comment to the next line, but will have comment.leading set to true
do.something(); // this is a comment, but will have comment.leading set to false
```